libyul/backends/wasm/WasmDialect.cpp

```diff
@@ -85,6 +87,69 @@ WasmDialect const& WasmDialect::instance()
 	return *dialect;
 }
 
+void WasmDialect::addEthereumExternals()
+{
+	// These are not YulStrings because that would be too complicated with regards
+	// to the YulStringRepository reset.
+	static string const i64{"i64"};
+	static string const i32{"i32"};
+	static string const i32ptr{"i32"}; // Uses "i32" on purpose.
+	struct External { string name; vector<string> parameters; vector<string> returns; };
+	static vector<External> externals{
+		{"getAddress", {i32ptr}, {}},
+		{"getExternalBalance", {i32ptr, i32ptr}, {}},
+		{"getBlockHash", {i64, i32ptr}, {i32}},
+		{"call", {i64, i32ptr, i32ptr, i32ptr, i32}, {i32}},
+		{"callDataCopy", {i32ptr, i32, i32}, {}},
+		{"getCallDataSize", {}, {i32}},
+		{"callCode", {i64, i32ptr, i32ptr, i32ptr, i32}, {i32}},
+		{"callDelegate", {i64, i32ptr, i32ptr, i32}, {i32}},
+		{"callStatic", {i64, i32ptr, i32ptr, i32}, {i32}},
+		{"storageStore", {i32ptr, i32ptr}, {}},
+		{"storageLoad", {i32ptr, i32ptr}, {}},
+		{"getCaller", {i32ptr}, {}},
+		{"getCallValue", {i32ptr}, {}},
```

Comment:
Why doesn't this have a return?

Reply:
If the return value is longer than 64 bits, it is stored in memory.

Reply:
oh ok